### PR TITLE
Simplify java.time.Duration creation units to be more readable

### PR DESCRIPTION
**Pull request by garyolsen**
As requested in #2687; tests should fully illustrate the behavior.

I'm taking a stab at the condition covered by the last test case since it seems doable, but putting up a PR for this since it should be good-to-go without it.

### CODE REVIEW

File: rewrite-java/src/main/java/org/openrewrite/java/cleanup/SimplifyDurationCreationUnits.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        DurationUnits(String methodName, TemporalUnit unit) {
+            this.methodName = methodName;
+            this.millisFactor = Duration.of(1, unit).toMillis();
+            this.methodMatcher = new MethodMatcher("java.time.Duration " + methodName + "(long)");
+        }
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Simplify java.time.Duration units";
```
Comment (sambsnyd) [inline rewrite-java/src/main/java/org/openrewrite/java/cleanup/SimplifyDurationCreationUnits.java:59]:
```suggestion
        return "Simplify `java.time.Duration` units";
```
Can use markdown in recipe titles and descriptions

File: rewrite-java/src/main/java/org/openrewrite/java/cleanup/SimplifyDurationCreationUnits.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        return "Simplifies java.time.Duration units to be more human-readable.";
+    }
+
+    @Override
+    protected @Nullable TreeVisitor<?, ExecutionContext> getApplicableTest() {
+        TreeVisitor<?, ExecutionContext>[] tests = new TreeVisitor[DurationUnits.values().length];
+        for (int i = 0; i < tests.length; i++) {
+            tests[i] = new UsesMethod(DurationUnits.values()[i].methodMatcher);
+        }
+        return Applicability.or(tests);
```
Comment (sambsnyd) [inline rewrite-java/src/main/java/org/openrewrite/java/cleanup/SimplifyDurationCreationUnits.java:73]: Could probably be simplified with a single MethodMatcher for `Duration of*(long)`. If I remember correctly the wildcards work in method names as well

Comment (garyolsen) [reply rewrite-java/src/main/java/org/openrewrite/java/cleanup/SimplifyDurationCreationUnits.java:73]: I'm ignoring `ofNanos` etc, which is why it's like this, but on second though it's probably more efficient to use the wildcard.

Comment (garyolsen) [reply rewrite-java/src/main/java/org/openrewrite/java/cleanup/SimplifyDurationCreationUnits.java]: Done

**Review by sambsnyd — APPROVED** [review]

**Pull request merged**